src/Parse/ParseObject.php

```diff
@@ -1367,4 +1369,18 @@ public static function query()
             return new ParseQuery($subclass);
         }
     }
+
+    /**
+     * Check if array is Associative
+     * Array is not associative is keys are numeric
+     *
+     * @return bool
+     */
+    private static function isAssoc($arr)
```

Comment:
You can remove this static method altogether. Although your solution does work, it's more work than is needed; given that it was just a lack of using the proper array keys.